---
float32[] colorImage
float32[] depthImage